File: Clases2025-26/Clase08-RecyclerView/app/src/main/java/com/example/clase08_recyclerview/UsuarioAdapter.java
package com.example.clase08_recyclerview;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.clase08_recyclerview.databinding.UsuarioBinding;

import java.util.ArrayList;

public class UsuarioAdapter extends RecyclerView.Adapter<UsuarioAdapter.UsuarioViewHolder> {
    private ArrayList<Usuario> usuarios;
    public UsuarioAdapter(ArrayList<Usuario> usuarios) {
        this.usuarios = usuarios;
    }

    @NonNull
    @Override
    public UsuarioAdapter.UsuarioViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        UsuarioBinding binding=UsuarioBinding.inflate(LayoutInflater.from(parent.getContext()),parent,false);
        return new UsuarioViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull UsuarioAdapter.UsuarioViewHolder holder, int position) {
        Usuario usuario=usuarios.get(position);
        holder.bind(usuario);
    }

    @Override
    public int getItemCount() {
        return usuarios.size();
    }

    public class UsuarioViewHolder  extends RecyclerView.ViewHolder{

        private UsuarioBinding binding;
        public UsuarioViewHolder(UsuarioBinding binding) {
            super(binding.getRoot());
            this.binding=binding;

        }

        public void bind(Usuario usuario){
            binding.setUser(usuario);
        }




    }
}

File: Clases2023-24/Clase19-DudasFinal/PrietoNoel/app/src/main/java/com/example/prietonoel/UsuarioAdapter.kt
package com.example.prietonoel

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AlertDialog
import androidx.core.content.ContextCompat.startActivity
import androidx.recyclerview.widget.RecyclerView
import com.example.prietonoel.databinding.DatoUsuarioBinding

class UsuarioAdapter(var datos: ArrayList<Usuario>) :
    RecyclerView.Adapter<UsuarioAdapter.UsuariosViewHolder>() {

    class UsuariosViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var binding = DatoUsuarioBinding.bind(view)
        fun bind(usuario: Usuario) {
            binding.user = usuario
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UsuariosViewHolder {
        val inflater =
            LayoutInflater.from(parent.context).inflate(R.layout.dato_usuario, parent, false)
        return UsuariosViewHolder(inflater.rootView)
    }

    override fun getItemCount(): Int = datos.size

    override fun onBindViewHolder(holder: UsuariosViewHolder, position: Int) {
        var usuario = datos[position]
        holder.bind(usuario)
        holder.itemView.setOnClickListener {
            var builder = AlertDialog.Builder(holder.itemView.context)
            builder.setTitle("Confirmacion")
            builder.setMessage("¿Seguro que desea visualizar?")
            builder.setPositiveButton("Aceptar") { dialog, _ ->
                //lanzar la posicion
                val intent = Intent(holder.itemView.context, MainActivity::class.java)
                var bundle = Bundle().apply{
                    putInt("posicion", position)
                }
                intent.putExtra("posicion", bundle)
                startActivity(holder.itemView.context, intent, bundle)
                dialog.dismiss()
            }
            builder.setNegativeButton("Cancelar") { dialog, _ ->
                dialog.cancel()
            }
            builder.show()
        }
    }

}